src/aca_main.cpp

```diff
@@ -263,17 +285,39 @@ int main(int argc, char *argv[])
     aca_cleanup();
     return rc;
   }
-  // monitor br-int for dhcp request message
-  ovs_monitor_brint_thread =
-          new thread(bind(&ACA_OVS_Control::monitor,
-                          &ACA_OVS_Control::get_instance(), "br-int", "resume"));
-  ovs_monitor_brint_thread->detach();
 
-  // monitor br-tun for arp request message
-  ACA_OVS_Control::get_instance().monitor("br-tun", "resume");
+  // get bridge and dpid mappings from ovs
+  std::unordered_map<uint64_t, std::string> switch_dpid_map =
+          aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().get_ovs_bridge_mapping();
+
+  // get system port name and ofportid mappings from ovs
+  std::unordered_map<std::string, std::string> port_id_map =
+          aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().get_system_port_ids();
+
+  // set bridge controller will clean up flows
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().setup_ovs_controller(g_ovs_ctrl_address, g_ovs_ctrl_port);
+
+  // start local ovs server (openflow controller)
+  g_ovs_ctrl = new OFController(switch_dpid_map, port_id_map, g_ovs_ctrl_address.c_str(), g_ovs_ctrl_port);
+  g_ovs_ctrl->start();
+
+  // pass ovs_ctrl handle to l2 programmer
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().set_openflow_controller(g_ovs_ctrl);
```

Comment:
@lfu-ps It seems like we can put all these statements in a dedicated function (such as `setup_openflow` in `ACA_OVS_L2_Programmer`, then we can make only 1 call here.

Reply:
Fixed and resolved.